docs/algorithms/libfuzzer/implementation_en.md

```diff
@@ -0,0 +1,120 @@
+
+### Mutator
+
+#### Mutate_AddWordFromTORC (CMP)
+
+The original libFuzzer adds this mutator when option CMP is enabled. It records the information of branching by comparison operations and performs mutation.
+If the compiler has the `-fsanitize-coverage=trace-cmp` option, the compiler will instrument the executable to record the comparison target and operator type at runtime when the executable branches on a comparison operation. CMP will look for a value in the input that is the same as the value used in the comparison and rewrites it so that the conditional branch goes into a different side. This mutation is based on the following idea: even if the input has the same value as the value used in the comparison, it does not guarantee the same value as the comparison target, but the fuzzer expects to change the branch direction with some probability.
+This mutator is unimplemented in fuzzuf because it does not support `trace-cmp` to get information about comparison operations.
+
+#### Mutate_AddWordFromPersistentAutoDictionary
+
+libFuzzer has a dictionary that stores the values inserted by `Mutate_AddWordFromManualDictionary` and `Mutate_AddWordFromTORC` that led to the discovery of new coverage. Since CMP is unimplemented in fuzzuf, this mutator only stores input from `Mutate_AddWordFromManualDictionary`.
+
+#### Custom Mutator
+
+libFuzzer provides CustomMutator and CustomCrossOver to add your mutators. fuzzuf does not provide nodes for these mutators because it is easier to add nodes by yourself than provide special nodes.
+
+### Corpus
+
+libFuzzer can persist the complete state of corpus and resume fuzzing. Still, the implementation of fuzzuf only persists the input values, so it cannot completely restore the previous state even if it resumes from the persistent information.
+
+### Feature
+
+#### Data Flow Trace
+
+Data Flow Trace uses a record of data movement obtained using LLVM's DataFlowSanitizer to determine which parts of the input values affect the branch. By masking the range of mutation based on this result, we can concentrate on finding the input that exits a particular branch. However, the original implementation of libFuzzer does not use this information to generate the mask.
+This feature is not implemented in fuzzuf because there is no way to get the information equivalent to DataFlowSanitizer similar to CMP.
+
+### Executor
+
+#### Avoiding Child Process Creation
+
+libFuzzer avoids the cost of creating child processes by linking the fuzzing target and the fuzzer into the same binary. Still, since there is no equivalent executor in fuzzuf, the implementation of fuzzuf creates child processes.
+
+#### Support for Shared Libraries
+
+libFuzzer has a mechanism to collect and combine the edge coverage of executable binaries and the edge coverage of shared libraries linked to them. This feature is not implemented in fuzzuf because there is no way to get coverage from shared libraries.
```

Comment:
```suggestion
libFuzzer has a mechanism to collect and combine the edge coverage of executable binaries and that of shared libraries linked to them. This feature is not implemented in fuzzuf because there is no way to get coverage from shared libraries.
```

Reply:
```suggestion
libFuzzer has a mechanism to collect and combine the edge coverage of both executable binaries and shared libraries linked to them. This feature is not implemented in fuzzuf because there is no way to get coverage from shared libraries.
```
To avoid the duplicated "the edge coverage" words, how about this fix?